### PR TITLE
Update Contract to Use Little Endian And Encoding Across Entire Repo

### PR DESCRIPTION
**Pull request by rauljordan**
This PR incorporates little endian encoding throughout our repo to match the Ethereum 2.0 spec.
Additionally, it adds more test support for encoding block deposits inputs accordingly.

Additionally, this PR updates our deposit contract to the latest version as of commit hash [d49e182775a11478ce6d8274be63a7c0822d9883](https://github.com/ethereum/deposit_contract/tree/d49e182775a11478ce6d8274be63a7c0822d9883) in the official Ethereum deposit contract repo for Ethereum 2.0. This maintains our structure of having a constructor to initialize values such as the max/min deposit amounts and the amount of deposits required for chain start.

The reason the two features are coupled into one PR is that tests across the repo depend on the contract being in big endian format, therefore an update to encoding across the repo would require an update to the contract itself as well.

This PR also includes a `skipChainstartDelay` constructor argument which tells the contract to fire chain start as soon as the deposit threshold is reached instead of waiting for a full day to do so as will be the case in the production contract.

### CODE REVIEW

File: contracts/deposit-contract/depositContract.v.py [hunk 1]
```diff
@@ -21,15 +22,27 @@ def __init__(depositThreshold: uint256,minDeposit: uint256,maxDeposit: uint256):
...
     self.MIN_DEPOSIT_AMOUNT = minDeposit
     self.MAX_DEPOSIT_AMOUNT = maxDeposit
-    
     for i in range(31):
         self.zerohashes[i+1] = sha3(concat(self.zerohashes[i], self.zerohashes[i]))
         self.branch[i+1] = self.zerohashes[i+1]
 
 @public
 @constant
-def to_bytes8(value: uint256) -> bytes[8]:
```
Comment (nisdas) [inline contracts/deposit-contract/depositContract.v.py]: Why was this removed ?

File: contracts/deposit-contract/depositContract.v.py [hunk 2]
```diff
@@ -71,15 +88,12 @@ def deposit(deposit_input: bytes[512]):
...
 
     if deposit_amount == self.MAX_DEPOSIT_AMOUNT:
         self.full_deposit_count += 1
         if self.full_deposit_count == self.CHAIN_START_FULL_DEPOSIT_THRESHOLD:
-            # Temporarily commenting out the day boundary. 
-            #timestamp_day_boundary: uint256 = deposit_timestamp - deposit_timestamp % SECONDS_PER_DAY + SECONDS_PER_DAY
-            #log.ChainStart(self.get_deposit_root(), self.to_bytes(timestamp_day_boundary))
-            log.ChainStart(self.get_deposit_root(), self.to_bytes8(deposit_timestamp))
-            self.chainStarted = True
+            timestamp_day_boundary: uint256 = as_unitless_number(block.timestamp) - as_unitless_number(block.timestamp) % SECONDS_PER_DAY + SECONDS_PER_DAY
```
Comment (nisdas) [inline contracts/deposit-contract/depositContract.v.py]: Can you comment out the day boundry as before and use the `deposit_timestamp` instead. It was commented out before because it broke our tests and validator flow.

File: contracts/deposit-contract/depositContract.v.py [hunk 3]
```diff
@@ -71,15 +90,15 @@ def deposit(deposit_input: bytes[512]):
...
     if deposit_amount == self.MAX_DEPOSIT_AMOUNT:
         self.full_deposit_count += 1
         if self.full_deposit_count == self.CHAIN_START_FULL_DEPOSIT_THRESHOLD:
-            # Temporarily commenting out the day boundary. 
-            #timestamp_day_boundary: uint256 = deposit_timestamp - deposit_timestamp % SECONDS_PER_DAY + SECONDS_PER_DAY
-            #log.ChainStart(self.get_deposit_root(), self.to_bytes(timestamp_day_boundary))
-            log.ChainStart(self.get_deposit_root(), self.to_bytes8(deposit_timestamp))
-            self.chainStarted = True
+            if self.skip_chainstart_delay:
+                log.ChainStart(self.get_deposit_root(), self.to_bytes8(deposit_timestamp))
```
Comment (nisdas) [inline contracts/deposit-contract/depositContract.v.py]: Use little endian here for the timestamp

**Comment by rauljordan** [comment]
PTAL @nisdas
